utils/default-helpers/src/main/scala/pl/touk/nussknacker/engine/util/functions/collection.scala

```diff
@@ -74,7 +74,7 @@ trait CollectionUtils extends HideToString {
   @Documentation(description =
     "Returns a slice of the list starting with start index (inclusive) and ending at stop index (exclusive)"
   )
-  @GenericType(typingFunction = classOf[ListTyping])
+  @GenericType(typingFunction = classOf[ListOrCollectionTyping])
```

Comment:
1. Why `ListTyping` here was causing an error?  Because `List` is too specific and given argument is a `Collection`?
2. Why List Or Collection? Can't we just check in this typing function that given "container" is a subclass of `Collection`?
3. Do we still use old typing functions somewhere or we can remove them?